New module: installp for AIX packages

kairoaraujo opened the pull request:
##### SUMMARY
This module works with installp AIX packages.
With this module is possible to manage (install and remove) native
AIX packages using simple playbooks.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
module/packaging/os/installp

##### ANSIBLE VERSION
```
ansible 2.5.0 (os_installp_aix e7fd8f36f3) last updated 2017/09/12 21:42:57 (GMT +200)
  config file = None
  configured module search path = ['/Users/kairoaraujo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/kairoaraujo/Dev/aix_installp/ansible/lib/ansible
  executable location = /Users/kairoaraujo/Dev/aix_installp/ansible/bin/ansible
  python version = 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06) [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```


##### ADDITIONAL INFORMATION
None

kairoaraujo committed:
Included module installp for AIX packages

This module works with installp AIX packages.
With this module is possible to manage (install and remove) native
AIX packages using simple playbooks.

kairoaraujo committed:
Included module installp for AIX packages

This module works with installp AIX packages.
With this module is possible manage (install and remove) native
AIX packages using simple playbooks.

kairoaraujo committed:
Merge branch 'os_installp_aix' of http://github.com/kairoaraujo/ansible into os_installp_aix

kairoaraujo committed:
Merge branch 'os_installp_aix' of http://github.com/kairoaraujo/ansible into os_installp_aix

kairoaraujo committed:
Merge branch 'os_installp_aix' of http://github.com/kairoaraujo/ansible into os_installp_aix

Shaps commented on lib/ansible/modules/packaging/os/installp.py:
```diff
@@ -0,0 +1,195 @@
+                                choices=["preview", "install"]),
+        ))
+
+    packages = module.params
```
I wouldn't call `module.params` as `packages` since it will hold also other parameters, not only the packages

Shaps commented on lib/ansible/modules/packaging/os/installp.py:
```diff
@@ -0,0 +1,195 @@
+
+    packages = module.params
+
+    pkgs = packages["name"].split(",")
```
If you want to use a comma-separated list of package names, you should mention it in `DOCUMENTATION` and the `EXAMPLES`

Shaps reviewed (changes requested):
Also, please squash your commits to remove the `Merge` commits, as the PR won't be merged otherwise.
https://github.com/ansible/ansible/blob/devel/CODING_GUIDELINES.md#git-practices

kairoaraujo committed:
Merge branch 'devel' of https://github.com/ansible/ansible into os_installp_aix

kairoaraujo committed:
Fixed variable for installp parameters and docs

Followed the comments for merge
- changed variable for installp params
- doc for name and multiple packges using comma

abadger commented on lib/ansible/modules/packaging/os/installp.py:
```diff
@@ -0,0 +1,194 @@
+        ))
+
+    installp_params = module.params
+    pkgs = installp_params["name"].split(",")
```
Rather than manually split here, you should define this as a list in the argument_spec:

``` python
            name=dict(aliases=["pkg"], required=True, type='list'),
```

An argument_spec list can either be handed a yaml list or it can receive a comma separated string (which it will automatically split into a list)

kairoaraujo committed:
Replaced manual split to a list according argument_spec

Changed manual split for module.params["name"] to argument_spec

kairoaraujo committed:
Fixed PEP8 space after operator

Fixed PEP8 space after operator

kairoaraujo committed:
improved package verification and pep8

- improved package/fileset verification on AIX before to take
actions to remove or install
- Applied PEP8 and non-written conventions
- better comments and documentation

note: even If the package is already installed, even the
package/fileset is new, the module will not install it.
On AIX it is not recommended because it can break your
OS version with missen filesets.

kairoaraujo committed:
applied single/double quotes convention.

applied single/double quotes convention.

kairoaraujo committed:
Package name 'all' allowed as AIX command line

- Allowed use 'name'  as all such as AIX allows
- Removed 'global' varible regarding licence that is used
only by install()

kairoaraujo committed:
Documentation format issue

E302 DOCUMENTATION is not valid YAM

gforster reviewed (commented):
curious on the status of this. looks like all is good?

gforster commented:
bot_status

gforster commented on lib/ansible/modules/packaging/os/installp.py:
```diff
@@ -0,0 +1,347 @@
+short_description: Installing packages for AIX
+description:
+  - Manage packages for AIX using 'installp'.
+version_added: "2.5"
```
```suggestion
version_added: "2.8"
```
Should this be re-targeted for 2.8?

dagwieers replied:
Fixed.

gforster commented:
Should this be renamed to aix_installp? I'm not sure that installp exists anywhere outside of AIX, but the aix_prefix may still be desired.

bcoca commented:
note: unrelated to this module being accepted, we should add installp to fact gathering to identify ansible_pkg_mgr correctly.

dagwieers commented:
> Should this be renamed to aix_installp? I'm not sure that installp exists anywhere outside of AIX, but the aix_prefix may still be desired.

It is not a requirement, most of the packaging modules are not prefixed. I do think it would be beneficial to move all AIX-specific modules to its own *aix/* sub-directory. But would only do it once we have everything merged and all related issues closed.

dagwieers committed:
Required (cosmetic) changes and improvements

dagwieers committed:
Small changes

dagwieers committed:
Silly PEP8 fixes

dagwieers committed:
Fix underscore variable
